src/components/Editor/Footer.js

```diff
@@ -172,8 +193,13 @@ export default connect(
   state => {
     const selectedSource = getSelectedSource(state);
     const selectedId = selectedSource && selectedSource.get("id");
+    const source = selectedSource.toJS();
+    const mappedSource = isOriginal(source)
+      ? getGeneratedSource(state, source)
```

Comment:
might not need the check here and just do `const mappedSource = getGeneratedSource(state, source)` since we already do a check inside `getGeneratedSource`. (This comment depends on  what is returned if there is no source)

Reply:
done

Reply:
turned out we need to check here cause it would break if source is not mapped